Loop/Managers/DeviceDataManager.swift

```diff
@@ -417,27 +448,19 @@ class DeviceDataManager: CarbStoreDelegate, TransmitterDelegate, ReceiverDelegat
             }
         }
 
-        // If we don't have a recent reservoir value, or the pump was recently rewound, read a new reservoir value before bolusing.
-        if  latestReservoirValue == nil ||
+        // If we don't have recent pump data, or the pump was recently rewound, read new pump data before bolusing.
+        if  latestPumpStatusDate == nil ||
             latestReservoirVolumeDrop < 0 ||
-            latestReservoirValue!.startDate.timeIntervalSinceNow <= NSTimeInterval(minutes: -5)
+            latestPumpStatusDate!.timeIntervalSinceNow <= NSTimeInterval(minutes: -5)
         {
-            readReservoirVolume { (result) in
-                switch result {
-                case .Success(let (unitVolume, date)):
-                    self.doseStore.addReservoirValue(unitVolume, atDate: date) { (newValue, _, error) in
-                        if let error = error {
-                            self.logger.addError(error, fromSource: "Bolus")
-                            completion(error: LoopError.CommunicationError)
-                        } else {
-                            self.latestReservoirValue = newValue
-                            setBolus()
-                        }
-                    }
-                case .Failure(let error):
-                    completion(error: error)
+            readAndUpdatePumpData({ (error) in
```

Comment:
This code path, unfortunately, triggers a whole bunch of things like pump history fetching and the basal adjustment calculation via `updateReservoirVolume`, which is why the previous block wasn't calling it. I'm working on a few refactors to streamline this file.. sorry for the confusion.